File: frontend/src/service/budget_service.js
import axios from '../util/axios';

const API_URL = window.location.protocol + '//' + window.location.host + '/api/budget/';

class BudgetService {
    /**
     * 예산 목록 조회
     * @param {*} BudgetRequestDto
     * @returns
     */
    selectBudgetList(BudgetRequestDto) {
        return axios
            .post(API_URL + 'list', BudgetRequestDto)
            .then(response => {
                return response.data;
            });
    }
    /**
     * 예산 목록 저장
     * @param {*} BudgetDto
     * @returns
     */
    saveBudgetList(BudgetDto) {
        return axios
            .post(API_URL + 'save', BudgetDto)
            .then(response => {
                return response.data;
            });
    }
    /**
     * 예산 대비 지출 목록 조회
     * @param {*} BudgetRequestDto
     * @returns
     */
    selectBudgetExpenditureList(BudgetRequestDto) {
        return axios
            .post(API_URL + 'expenditure/list', BudgetRequestDto)
            .then(response => {
                return response.data;
            });
    }


}

export default new BudgetService();

File: src/main/java/com/app/service/BudgetService.java
package com.app.service;

import com.app.dto.BudgetDto;
import com.app.dto.BudgetExpenditureDto;
import com.app.dto.BudgetRequestDto;

import java.util.List;

public interface BudgetService {
    /**
     * 예산 목록 조회
     * @param budgetRequestDto
     * @return BudgetDto
     */
    BudgetDto selectBudgetList(BudgetRequestDto budgetRequestDto);

    /**
     * 예산 목록 저장
     * @param budgetDto
     * @return void
     */
    void saveBudgetList(BudgetDto budgetDto);

    /**
     * 예산 대비 지출 목록 조회
     * @param budgetRequestDto
     * @return List<BudgetExpenditureDto>
     */
    List<BudgetExpenditureDto> selectBudgetExpenditureList(BudgetRequestDto budgetRequestDto);
}
